### PR TITLE
barrows plugin: add a toggle for overlay

### LINKED ISSUE #7046: [Suggestion] Add a toggle for the Barrows info box (vanilla one is better imo)
https://i.gyazo.com/75eca0740e8305839ecb520049d43db0.png

### PR DESCRIPTION
**Pull request by jwngo**
Closes #7046

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/barrows/BarrowsBrotherSlainOverlay.java [hunk 1]
```diff
@@ -54,41 +57,44 @@ private BarrowsBrotherSlainOverlay(Client client)
 	@Override
 	public Dimension render(Graphics2D graphics)
 	{
-		// Do not display overlay if potential is null/hidden
-		final Widget potential = client.getWidget(WidgetInfo.BARROWS_POTENTIAL);
-		if (potential == null || potential.isHidden())
+		if (config.showBarrowsOverlay())
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/barrows/BarrowsBrotherSlainOverlay.java]: This should be a guard clause, not a large nested if statement.

Comment (jwngo) [reply runelite-client/src/main/java/net/runelite/client/plugins/barrows/BarrowsBrotherSlainOverlay.java]: Changed it to guard clause.
Not sure whether the default barrows overlay can become movable, just adding toggle as requested by #7046.

File: runelite-client/src/main/java/net/runelite/client/plugins/barrows/BarrowsBrotherSlainOverlay.java [hunk 2]
```diff
@@ -89,6 +91,10 @@ public Dimension render(Graphics2D graphics)
 				.rightColor(rewardPercent >= 73.0f && rewardPercent <= 88.0f ? Color.GREEN : rewardPercent < 65.6f ? Color.WHITE : Color.YELLOW)
 				.build());
 
-		return panelComponent.render(graphics);
+		// Toggle for barrows overlay
+		if (config.showBarrowsOverlay())
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/barrows/BarrowsBrotherSlainOverlay.java]: This isn't the right way to do a guard clause; we shouldn't even be running the code that updates `panelComponent` if `config.showBarrowsOverlay()` is false.

**Comment by Adam-** [comment]
I think we were discussing just using the in-game interface and making it movable, which is preferable.

**Pull request closed without merging**